### DISCUSSION TITLE
Collapsing behavior improvements

### DISCUSSION THREAD

**Discussion by hankbinkle** [phcode-dev/phoenix#1033] [Ideas]
1. Collapse All does not seem to do what you would want.  It collapses all the way to the top indent, usually the <html> tag, which is great. But when you start expanding sub-elements, not all of them are collapsed as well, meaning Collapse All did not really do anything useful. It would be great if Collapse All would collapse all of the elements, as the name implies.
2. Some Intermediate Collapsing would be useful. Like Collapse 2, 3, 4, or n levels.
3. If the cursor is on a collapsed element, Collapse Current does not do anything.  You would expect it to Collapse the parent element, allowing you to collapse up the tree with successive Collapse Current executions.
4. Remember Collapse would be amazing so you don't have to reset where you were the last time you opened/were working on a document!
5. Collapsing does not seem to work with Template literals/multi-line strings.

**Comment by abose** [comment]
@hankbinkle Thanks for the detailed report. I agree with you on the desired behavior of the folds. Have created issue to track: https://github.com/phcode-dev/phoenix/issues/1034

The `collapse all` behavior is fixed in phcode. The other issues might take a bit more time to get resolved.

**Comment by hankbinkle** [comment]
wow! awesome!

Thanks,

Rob

Rob Zinn
c:904.377.9854
***@***.***

Sent via Superhuman iOS ***@***.***>
